ICO_Template/ICO_Template.cs

```diff
@@ -121,8 +121,16 @@ public static bool MintTokens()
             // 众筹成功
             ulong token = value * swap_rate / 100000000;
             BigInteger balance = Storage.Get(Storage.CurrentContext, sender).AsBigInteger();
-            Storage.Put(Storage.CurrentContext, sender, token + balance);
             BigInteger totalSupply = Storage.Get(Storage.CurrentContext, "totalSupply").AsBigInteger();
+
+            if (totalSupply + token > total_amount)
```

Comment:
by the last time ,  part of the NEO could be success.  like below
`            // ico_neo - the target of ico neo amount 
            BigInteger ico_neo = Storage.Get(Storage.CurrentContext, "icoNeo").AsBigInteger();
            // balance_neo - the amount of this time, total_neo - the neo had received,
            BigInteger balance_neo = total_neo - ico_neo;
            if (total_neo <= ico_neo)
            {
                Refund(sender, value);
                return false;
            } else if (balance_neo <= value)
            {
                Refund(sender, value - balance_neo);
                ico_neo += balance_neo;
                value = (ulong) balance_neo;
            } else
            {
                ico_neo += value;
            }
            Storage.Put(Storage.CurrentContext, "icoNeo", ico_neo);
            // crowdfunding success
            ulong token = value / neo_decimals * swap_rate;
            Storage.Put(Storage.CurrentContext, sender, token);
            BigInteger totalSupply = Storage.Get(Storage.CurrentContext, "totalSupply").AsBigInteger();
            Storage.Put(Storage.CurrentContext, "totalSupply", token + totalSupply);
            Transferred(null, sender, token);`